src/psyclone/transformations.py

```diff
@@ -313,12 +313,161 @@ def apply(self, node, options=None):
         return schedule, keep
 
 
+class OMPTaskloopTrans(ParallelLoopTrans):
+
+    '''
+    Adds an OpenMP taskloop directive to a loop.
+
+    TODO: #1364 Taskloops do not yet support reduction clauses.
+
+    For example:
+
+    >>> from pysclone.parse.algorithm import parse
+    >>> from psyclone.psyGen import PSyFactory
+    >>> api = "gocean1.0"
+    >>> filename = "nemolite2d_alg.f90"
+    >>> ast, invokeInfo = parse(filename, api=api, invoke_name="invoke")
+    >>> psy = PSyFactory(api).create(invokeInfo)
+    >>>
+    >>> from psyclone.transformations import OMPParallelTrans, OMPSingleTrans
+    >>> from psyclone.transformations import OMPTaskloopTrans
+    >>> singletrans = OMPSingleTrans()
+    >>> paralleltrans = OMPParallelTrans()
+    >>> tasklooptrans = OMPTaskloopTrans()
+    >>>
+    >>> schedule = psy.invokes.get('invoke_0').schedule
+    >>> schedule.view()
+    >>>
+    # Apply the OpenMP Taskloop transformation to *every* loop
+    # in the schedule.
+    # FIXME: This ignores loop dependencies
+    >>> for child in schedule.children:
+    >>>     tasklooptrans.apply(child)
+    >>> # Enclose all of these loops within a single OpenMP
+    >>> # SINGLE region
+    >>> singletrans.apply(schedule.children)
+    >>> # Enclose all of these loops within a single OpenMP
+    >>> # PARALLEL region
+    >>> paralleltrans.apply(schedule.children)
+    >>> schedule.view()
+
+    '''
+    def __init__(self, grainsize=None, num_tasks=None):
+        self._grainsize = None
+        self._num_tasks = None
+        if grainsize is not None:
```

Comment:
Now that the setters accept None, I think the two if statements can be removed.

Reply:
Done

Reply:
Actually this highlighted a bug I'd missed which is now fixed.